batch-operations/src/main/java/com/hartwig/batch/operations/SageGermline.java

```diff
@@ -31,73 +29,66 @@
+        // Download normal
+        commands.addCommand(() -> remoteReferenceFile.toCommandForm(localReferenceFile));
+        commands.addCommand(() -> remoteReferenceIndex.toCommandForm(localFilename(remoteReferenceIndex)));
 
-        // Convert to bam if necessary
-        if (!localReferenceFile.equals(localReferenceBam)) {
-            startupScript.addCommands(cramToBam(localReferenceFile));
-        }
+        final SageCommandBuilder sageCommandBuilder = new SageCommandBuilder(resourceFiles).germlineMode(referenceSampleName,
+                localReferenceFile,
+                tumorSampleName,
+                localTumorFile);
 
-        // Run post processing (NONE for germline)
-        final SubStageInputOutput postProcessing = sageApplication.apply(SubStageInputOutput.empty(referenceSampleName));
-        startupScript.addCommands(postProcessing.bash());
+        SageApplication sageApplication = new SageApplication(sageCommandBuilder);
+        SagePostProcessGermline sagePostProcess = new SagePostProcessGermline(referenceSampleName, tumorSampleName, resourceFiles);
+        SubStageInputOutput sageOutput = sageApplication.andThen(sagePostProcess).apply(SubStageInputOutput.empty(tumorSampleName));
+        commands.addCommands(sageOutput.bash());
 
         // Store output
-        startupScript.addCommand(new OutputUpload(GoogleStorageLocation.of(runtimeBucket.name(), "sage"), executionFlags));
-
-        return VirtualMachineJobDefinition.sageCalling(startupScript, ResultsDirectory.defaultDirectory());
-    }
+        commands.addCommand(new OutputUpload(GoogleStorageLocation.of(runtimeBucket.name(), "sage"), executionFlags));
 
-    private List<BashCommand> cramToBam(String cram) {
-
-        final String output = cram.replace("cram", "bam");
-        final BashCommand toBam = new VersionedToolCommand("samtools",
-                "samtools",
-                Versions.SAMTOOLS,
-                "view",
-                "-o",
-                output,
-                "-O",
-                "bam",
-                "-@",
-                Bash.allCpus(),
-                cram);
-
-        final BashCommand index =
-                new VersionedToolCommand("samtools", "samtools", Versions.SAMTOOLS, "index", "-@", Bash.allCpus(), output);
-
-        return Lists.newArrayList(toBam, index);
+        return VirtualMachineJobDefinition.sageCalling(commands, ResultsDirectory.defaultDirectory());
     }
 
     @Override
     public OperationDescriptor descriptor() {
-        return OperationDescriptor.of("SageGermline", "Generate germline output", OperationDescriptor.InputType.JSON);
+        return OperationDescriptor.of("SageGermline", "Generate sage output", OperationDescriptor.InputType.JSON);
+    }
+
+    private static String localFilename(InputFileDescriptor remote) {
+        return format("%s/%s", VmDirectories.INPUT, new File(remote.inputValue()).getName());
+    }
```

Comment:
We already have the `localDestination` method on `InputFileDescriptor` that does this I think.

Reply:
Thanks. I was the one who put it there :)